tests/track/params_test.py

```diff
@@ -2772,3 +2772,41 @@ def test_force_merge_all_params(self):
         assert p["request-timeout"] == 30
         assert p["max-num-segments"] == 1
         assert p["mode"] == "polling"
+
+
+class TestDownsampleParamSource:
+    def test_downsample_all_params(self):
+        source = params.DownsampleParamSource(
+            track.Track(name="unit-test"),
+            params={"source-index": "test-source-index", "target-index": "test-target-index", "fixed-interval": "1m"},
+        )
+
+        p = source.params()
+
+        assert p["fixed-interval"] == "1m"
+        assert p["source-index"] == "test-source-index"
+        assert p["target-index"] == "test-target-index"
+
+    def test_downsample_target_index_param(self):
+        source = params.DownsampleParamSource(
+            track.Track(name="unit-test"),
+            params={"source-index": "tsdb", "fixed-interval": "1m"},
+        )
+
+        p = source.params()
+
+        assert p["fixed-interval"] == "1m"
+        assert p["source-index"] == "tsdb"
+        assert p["target-index"] == "tsdb-1m"
+
+    def test_downsample_empty_params(self):
+        source = params.DownsampleParamSource(
+            track.Track(name="unit-test"),
+            params={},
+        )
+
+        p = source.params()
+
+        assert p["fixed-interval"] != ""
```

Comment:
This should be the documented default, and should need changing if the default changes